### DISCUSSION TITLE
is it possible to make git commit hash available for an app?

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#20528] [Q&A General] [answered]
Dear all, 
would it be possible  to have an app to console output the version and the Git commit hash (including possibly whether the repository was dirty) in a  similar to what can be done to the framework? 
So far, myapp --version it only shows the framework commit hash. However, it works if I directly force the git commit hash (like >git describe --always --dirty --match 'NOT A TAG').  
Thanks in advance,
Mauro

**Comment by mcacace** [comment, marked as answer]
I kind of found a solution by modifying the makefile of my application to include the commit stage:

-  GIT_VERSION := $(shell git --no-pager describe --tags --always --dirty)
- ADDITIONAL_CPPFLAGS += -DGOLEM_VERSION=\"$(GIT_VERSION)\"

And then by overwrite the getVersion() call in my base App.C to return my GOLEM_VERSION instead of the default MOOSE_VERSION. 

Now, my question: do you think it is the proper way or should I have done things differently?

Thanks for any feedback,
mauro

**Reply by cpgr** [reply]
Hey @mcacace 

I have done this before. From memory there is a `GEN_REVISION := no` in the app Makefile - changing this to a `yes` should generate a file `include/base/GolemRevision.h` that defines a variable `GOLEM_REVISION` that is the latest git hash.

So you can just add this header (`#include "GolemRevision"` ) wherever you need this information.
